tchannel/tornado/peer.py

```diff
@@ -583,7 +628,9 @@ def get(self, hostport):
         """
         assert hostport, "hostport is required"
         if hostport not in self._peers:
-            self._peers[hostport] = self.peer_class(self.tchannel, hostport)
+            self.add(hostport)
+            return self.get(hostport)
```

Comment:
Unnecessary recursive call. Change this to

    if hostport not in self._peers:
        self.add(hostport)
    return self._peers[hostport]